js/theme.js

```diff
@@ -339,6 +343,22 @@ class Theme {
             } else if (arrow.themedIconMode == "coursesDropDown") {
                 arrow.setAttribute("style", `background-image: url(${sourceUrl}); background-size: contain;`);
             }
+
+            // ***** Tracking courses missing icons
+            if (Setting.getValue("toggle_stopTrackingMissingIcons")) continue;

```

Comment:
This is partly a problem that the loop is huge, but the jump in control flow here seems unnecessary and introduces confusion, in my opinion. `if (!Setting.getValue(...)) { ... }` certainly wouldn't be the biggest/worst conditional block in this method.